google-cloud-spanner/src/main/java/com/google/cloud/spanner/Type.java

```diff
@@ -209,6 +228,15 @@ private Type(
     this.structFields = structFields;
   }
 
+  private Type(
+      Code code,
+      @Nullable Type arrayElementType,
+      @Nullable ImmutableList<StructField> structFields,
+      @Nullable String protoTypeFqn) {
```

Comment:
I would mark this as @NonNullable, as it is a constructor that should only be used for proto types.